File: 0387. First Unique Character in a String.java
// Easy
// Array
// Time:O(n), Space:O(1)
// https://leetcode.cn/problems/first-unique-character-in-a-string/

class Solution {
    public int firstUniqChar(String s) {
        int[] cnt = new int[26];
        for (char c : s.toCharArray()) {
            cnt[c - 'a']++;
        }

        for (int i = 0; i < s.length(); i++) {
            if (cnt[s.charAt(i) - 'a'] == 1)
                return i;
        }
        return -1;
    }
}

File: 0001-0500/0387. First Unique Character in a String.go
package main
// Easy
// Time:O(n), Space:O(1)
// https://leetcode.cn/problems/first-unique-character-in-a-string/

func firstUniqChar(s string) int {
    count := [26]int{}

    for _, c := range s {
        count[c - 'a']++
    }

    for i, c := range s {
        if count[c - 'a'] == 1 {
            return i
        }
    }
    return -1
}
